test/test_cabinet.py

```diff
@@ -118,38 +119,129 @@ def test_set_configuration_region_bad_protocol(self):
 
 
 @mock_s3
-class TestS3Cabinet(unittest.TestCase):
+@patch.dict(os.environ, {'AWS_ACCESS_KEY_ID': 'testing',
+                         'AWS_SECRET_ACCESS_KEY': 'testing',
+                         'AWS_SECURITY_TOKEN': 'testing',
+                         'AWS_SESSION_TOKEN': 'testing', })
+class TestS3CabinetNoRegion(unittest.TestCase):
 
     def setUp(self) -> None:
-        S3Cabinet.set_configuration(region_name='us-west-2')
+        self.client = None
+        self._bucket = 'mock-bucket'
+
+    def tearDown(self) -> None:
+        if self.client:
+            self.client.delete_bucket(Bucket=self._bucket)
 
     def test_set_configuration_region(self):
         self.assertIsNotNone(S3Cabinet.client)
-        self.assertEqual(S3Cabinet.client.meta.region_name, 'us-west-2')
 
     def test_read_create_s3_cabinet(self):
-        client = boto3.client('s3', region_name='us-west-2')
-        bucket = 'mock-bucket'
-        client.create_bucket(Bucket=bucket)
-        filename = f'{bucket}/test.yml'
+        self.client = boto3.client('s3')
```

Comment:
Might make sense to make this an instance variables on the test class

Reply:
The reason I didn't do this is because I wanted to test instantiating this test client with different region parameters.